src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -1,15 +1,51 @@
 package ch.fhnw.algd1.converters.binary;
 
 public class BinConverter {
-	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
-	}
-
-	public static int fromString(String text) {
-		// TODO expect text to contain 8 binary digits, parse to int value in
-		// 2-complement
-		return 0;
-	}
+public static String toString(int x) {
+
+/*
+    var str = Integer.toBinaryString(x & 0xFF); // it's now unsigned
+
+    var zeros = new String(new char[8 - str.length()]).replace('\0', '0');
+
+    return zeros + str;
+    */
+    
+ 
+    StringBuilder value = new StringBuilder();
+    if (x < 0) x += 256;
+    
+    int i = 1 << 7;
+    while (i >= 1) {
+        if (x >= i) {
+            value.append("1");
+            x -= i;
+        } else {
+            value.append("0");
+        }
+        i /= 2;
+    }
+    return value.toString();
+}
+
+public static int fromString(String text) {
+    int value = 1;
+    int result = 0;
+    char[] chars = text.toCharArray();
+    for (int i = 0; i < chars.length; i++) {
+        if (chars[chars.length - 1 - i] == '0') {
+            value *= 2;
+            continue;
+        }
+        if (i + 1 == chars.length) {
+            value = -value;
+            result += value;
```

Comment:
```suggestion
            result -= value;
```
wäre einfacher, oder?

Reply:
and then directly return, as there is no need to get one more check from the for...

Reply:
Lieber nicht. Das macht die Struktur des Programms noch schlechter erkennbar, als sie jetzt schon ist. Die Zeitersparnis davon ist so klein, dass das kein guter Trade ist.
Interessanterweise adressiert das mein nächster Kommentar ja auch, bzw. löst der gerade zwei Probleme: Vor allem wird die Struktur der Ausführungsreihenfolge auch im Programmtext sichtbar und nebenbei spart es nicht nur den letzten Test in der `for`-Anweisung, sondern auch noch den in Ihrer `if`-Anweisung.